pkg/driver/driver_openstack.go

```diff
@@ -223,7 +269,8 @@ func (d *OpenStackDriver) Create() (string, string, error) {
 
 	if err != nil {
 		metrics.APIFailedRequestCount.With(prometheus.Labels{"provider": "openstack", "service": "nova"}).Inc()
-		return "", "", fmt.Errorf("error creating the server: %s", err)
+		volerr := volumes.Delete(cinder, volume.ID, volumes.DeleteOpts{Cascade: true}).ExtractErr()
```

Comment:
maybe add an check here if an id is present to avoid unnecessary api calls (if there is no volumeType no Image ID is present to